test/hotspot/gtest/utilities/test_globalCounter.cpp

```diff
@@ -24,93 +24,65 @@
+constexpr const int good_value = 1337;
+constexpr const int bad_value =  4711;
 
 struct TestData {
   long test_value;
 };
 
-class RCUReaderThread : public JavaTestThread {
-public:
-  static volatile bool _exit;
-  volatile TestData** _test;
-  Semaphore* _wrt_start;
-  RCUReaderThread(Semaphore* post, volatile TestData** test, Semaphore* wrt_start)
-    : JavaTestThread(post), _test(test), _wrt_start(wrt_start) {};
-  virtual ~RCUReaderThread(){}
-  void main_run() {
-    _wrt_start->signal();
-    while (!_exit) {
-      GlobalCounter::CSContext cs_context = GlobalCounter::critical_section_begin(this);
-      volatile TestData* test = Atomic::load_acquire(_test);
-      long value = Atomic::load_acquire(&test->test_value);
-      ASSERT_EQ(value, GOOD_VALUE);
-      GlobalCounter::critical_section_end(this, cs_context);
+TEST_VM(GlobalCounter, critical_section) {
+  constexpr int number_of_readers = 4;
+  volatile bool rt_exit = false;
+  Semaphore wrt_start;
+  volatile TestData* test = nullptr;
+
+  auto rcu_reader = [&](Thread* current, volatile TestData** _test) {
+    wrt_start.signal();
+    while (!rt_exit) {
+      GlobalCounter::CSContext cs_context = GlobalCounter::critical_section_begin(current);
+      volatile TestData* read_test = Atomic::load_acquire(_test);
+      long value = Atomic::load_acquire(&read_test->test_value);
+      ASSERT_EQ(value, good_value);
+      GlobalCounter::critical_section_end(current, cs_context);
       {
-        GlobalCounter::CriticalSection cs(this);
+        GlobalCounter::CriticalSection cs(current);
         volatile TestData* test = Atomic::load_acquire(_test);
         long value = Atomic::load_acquire(&test->test_value);
-        ASSERT_EQ(value, GOOD_VALUE);
+        ASSERT_EQ(value, good_value);
       }
     }
-  }
-};
-
-volatile bool RCUReaderThread::_exit = false;
-
-class RCUWriterThread : public JavaTestThread {
-public:
-  RCUWriterThread(Semaphore* post) : JavaTestThread(post) {
   };
-  virtual ~RCUWriterThread(){}
-  void main_run() {
-    static const int NUMBER_OF_READERS = 4;
-    Semaphore post;
-    Semaphore wrt_start;
-    volatile TestData* test = NULL;
-
-    RCUReaderThread* reader1 = new RCUReaderThread(&post, &test, &wrt_start);
-    RCUReaderThread* reader2 = new RCUReaderThread(&post, &test, &wrt_start);
-    RCUReaderThread* reader3 = new RCUReaderThread(&post, &test, &wrt_start);
-    RCUReaderThread* reader4 = new RCUReaderThread(&post, &test, &wrt_start);
 
-    TestData* tmp = new TestData();
-    tmp->test_value = GOOD_VALUE;
-    Atomic::release_store_fence(&test, tmp);
+  TestThreadGroup<decltype(rcu_reader), volatile TestData**, number_of_readers> ttg(rcu_reader,
+                                                                                    [&]() {
+                                                                                      return &test;
+                                                                                    });
```

Comment:
Can you format the Symbol case like this?  makes it easier to parse the syntax and match the ()s.